### PR TITLE
Specify the result of integer cast on boolean

### LINKED ISSUE #39190: Document the behavior of `true|false as {integral}`
Currently, it appears that `true as {integral} == 1 && false as {integral} == 0`. Unfortunately, the only documentation on this subject appears to be a tangentially related bullet point in the [reference](https://doc.rust-lang.org/reference.html#behavior-considered-undefined).

It should probably be documented in the [the bool documentation](https://doc.rust-lang.org/std/primitive.bool.html).

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #39190.

r? @frewsxcv

### CODE REVIEW

File: src/libstd/primitive_docs.rs [hunk 1]
```diff
@@ -12,7 +12,8 @@
 //
 /// The boolean type.
 ///
-/// The `bool` represents a value, which could only be either `true` or `false`.
+/// The `bool` represents a value, which could only be either `true` or `false`. If you cast
+/// a `bool` into an integer, `true` will be 1 and `false` will be.
```
Comment (rkruppe) [inline src/libstd/primitive_docs.rs]: Missing the 0.

**Comment by GuillaumeGomez** [comment]
@rkruppe: Good catch, thanks!

**Comment by frewsxcv** [comment]
If you wanted to, you could also add an example in the `# Examples` section. Otherwise, r=me

**Comment by GuillaumeGomez** [comment]
I'll then, good idea!

**Comment by GuillaumeGomez** [comment]
Added examples as well.

**Pull request merged**